### PR TITLE
Cassandra java tests

### PR DESCRIPTION
**Pull request by aschugunov**

### CODE REVIEW

File: instrumentation/cassandra/cassandra-3.0/javaagent/src/test/java/CassandraClientTest.java [hunk 1]
```diff
@@ -0,0 +1,422 @@
...
+                          .hasKind(SpanKind.CLIENT)
+                          .hasNoParent()
+                          .hasAttributesSatisfying(
+                              attributes ->
+                                  assertThat(attributes)
+                                      .containsEntry(NET_SOCK_PEER_ADDR, "127.0.0.1")
+                                      .containsEntry(NET_SOCK_PEER_NAME, "localhost")
+                                      .containsEntry(NET_SOCK_PEER_PORT, cassandraPort)
+                                      .containsEntry(DB_SYSTEM, "cassandra")
+                                      .containsEntry(DB_STATEMENT, "USE " + parameter.keyspace))
```
Comment (trask) [inline instrumentation/cassandra/cassandra-3.0/javaagent/src/test/java/CassandraClientTest.java:100]: can you change the assertions with fixed number of attributes to use `hasAttributesSatisfyingExactly`?

**Review by trask — COMMENTED** [review]
thx @aschugunov!

File: instrumentation/cassandra/cassandra-3.0/javaagent/src/test/java/CassandraClientTest.java (hunk 1 above)
Comment (aschugunov) [reply instrumentation/cassandra/cassandra-3.0/javaagent/src/test/java/CassandraClientTest.java:100]:
```
.hasAttributesSatisfyingExactly(
      equalTo(DB_CASSANDRA_TABLE, null)
)
```
It works, in this case dividing is not necessary

Comment (aschugunov) [reply instrumentation/cassandra/cassandra-3.0/javaagent/src/test/java/CassandraClientTest.java]: @trask Could you check my changes, please?

File: instrumentation/cassandra/cassandra-3.0/javaagent/src/test/java/CassandraClientTest.java [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+                  span ->
+                      span.hasName("DB Query")
+                          .hasKind(SpanKind.CLIENT)
+                          .hasNoParent()
+                          .hasAttributesSatisfyingExactly(
+                              equalTo(NET_SOCK_PEER_ADDR, "127.0.0.1"),
+                              equalTo(NET_SOCK_PEER_NAME, "localhost"),
+                              equalTo(NET_SOCK_PEER_PORT, cassandraPort),
+                              equalTo(DB_SYSTEM, "cassandra"),
+                              equalTo(DB_NAME, null),
```
Comment (trask) [inline instrumentation/cassandra/cassandra-3.0/javaagent/src/test/java/CassandraClientTest.java:98]:
```suggestion
```

File: instrumentation/cassandra/cassandra-3.0/javaagent/src/test/java/CassandraClientTest.java [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+                              equalTo(DB_STATEMENT, "USE " + parameter.keyspace),
+                              equalTo(DB_OPERATION, null),
+                              equalTo(DB_CASSANDRA_TABLE, null))),
```
Comment (trask) [inline instrumentation/cassandra/cassandra-3.0/javaagent/src/test/java/CassandraClientTest.java:101]: same, you can remove the `null` verifications b/c `hasAttributesSatisyingExactly` will fail if there are any extra attributes that you didn't verify

**Review by trask — COMMENTED** [review]
@aschugunov sorry for the late reply, and thanks for the help with the groovy -> Java conversion!

File: instrumentation/cassandra/cassandra-3.0/javaagent/src/test/java/CassandraClientTest.java (hunk 3 above)
Comment (aschugunov) [reply instrumentation/cassandra/cassandra-3.0/javaagent/src/test/java/CassandraClientTest.java:101]: Sure, thanks for the advice

**Comment by aschugunov** [comment]
It looks like build failed due to connectivity problem and rebuild will help. 

```
Downloading https://services.gradle.org/distributions/gradle-7.5.1-bin.zip
Error: Exception in thread "main" java.net.SocketException: An established connection was aborted by the software in your host machine
	at java.base/sun.nio.ch.NioSocketImpl.implWrite(NioSocketImpl.java:420)
	at java.base/sun.nio.ch.NioSocketImpl.write(NioSocketImpl.java:440)```

**Review by trask — APPROVED** [review]
thanks!

**Pull request merged**